src/lexer.coffee

```diff
@@ -500,25 +503,30 @@ exports.Lexer = class Lexer
       [input, id, colon] = match
       origin = @token 'CSX_TAG', id, 1, id.length
       @token 'CALL_START', '('
-      @token '{', '{'
+      csxStart = @token '[', '['
       @ends.push tag: '/>', origin: origin, name: id
       @csxDepth++
       return id.length + 1
     else if csxTag = @atCSXTag()
       if @chunk[...2] is '/>'
         @pair '/>'
-        @token '}', '}', 0, 2
+        @token ']', ']', 0, 2
         @token 'CALL_END', ')', 0, 2
         @csxDepth--
         return 2
       else if firstChar is '{'
-        token = @token '(', '('
+        if prevChar is ':'
+          token = @token '(', '('
+          @csxObjAttribute = no
```

Comment:
Not sure what the right fix here is, but because this is limited to a single state, alternately nesting assignment and splats causes errors:

```
[stdin]:1:24: error: unexpected }
<A child={<B {...params} />} />
                       ^
```

I suppose that's what `@pair` is for, but that will currently throw if it doesn't match... It would be nice to avoid a new stack though...